src/Adapter/Cart/CartPresenter.php

```diff
@@ -198,8 +201,12 @@ public function addCustomizedData(array $products, Cart $cart)
         }, $products);
     }
 
-    public function present(Cart $cart)
+    public function present(ObjectModel $cart)
     {
+        if (!is_a($cart, 'Cart')) {
```

Comment:
I changed type hinting to ObjectModel, so you can pass an instance of CMS or Address :trollface: 
Anyway, I will remove all type hinting for this method.